### PR TITLE
Settings dialog

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

**Comment by arkpar** [comment]
image

File: alethzero/SettingsDialog.cpp [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+struct Category
+{
+	int index = -1;
+	QString displayName;
+	QIcon icon;
+	SettingsPage *widget;
+};
+
+class CategoryModel : public QAbstractListModel
```
Comment (chriseth) [inline alethzero/SettingsDialog.cpp:53]: Space

File: alethzero/SettingsDialog.cpp [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+	int rowCount(QModelIndex const& _parent = QModelIndex()) const;
+	QVariant data(QModelIndex const& _index, int _role = Qt::DisplayRole) const;
+	QList<Category> const& categories() const { return m_categories; }
+
+private:
+	QList<Category> m_categories;
+	QIcon m_emptyIcon;
+};
+
+CategoryModel::CategoryModel(QObject* _parent)
```
Comment (chriseth) [inline alethzero/SettingsDialog.cpp:68]: `:` should go to the previous line

File: alethzero/SettingsDialog.cpp [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+	: QAbstractListModel(_parent)
+{
+	QPixmap empty(categoryIconSize, categoryIconSize);
+	empty.fill(Qt::transparent);
+	m_emptyIcon = QIcon(empty);
+}
+
+int CategoryModel::rowCount(const QModelIndex &parent) const
```
Comment (chriseth) [inline alethzero/SettingsDialog.cpp:76]: Position of `const`

File: alethzero/SettingsDialog.cpp [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+{
+	return parent.isValid() ? 0 : m_categories.size();
+}
+
+QVariant CategoryModel::data(const QModelIndex &index, int role) const
```
Comment (chriseth) [inline alethzero/SettingsDialog.cpp:81]: Position of `const`

File: alethzero/SettingsDialog.cpp [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+{
+	switch (role) {
```
Comment (chriseth) [inline alethzero/SettingsDialog.cpp:83]: Position of `{` (also below)

File: alethzero/plugins/browser/Cors.ui [hunk 6]
```diff
@@ -1,119 +1,19 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <ui version="4.0">
  <class>Cors</class>
- <widget class="QDialog" name="Cors">
+ <widget class="QWidget" name="Cors">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>534</width>
```
Comment (chriseth) [inline alethzero/plugins/browser/Cors.ui:9]: If this is a pixel value, then this will be another dialog that does not display correctly on my screen ;-)

**Pull request merged**